### ISSUE TITLE
[DOTX031E] Error when xref points to topic containing xref in its short description.

### ISSUE THREAD

**Issue by shaneataylor** [dita-ot/dita-ot#2708]
This is related to, but I think distinct from, #1868 .
# ENVIRONMENT
DITA-OT 2.3.1, MacOS Sierra (10.12)

# INPUT
```
<map>
  <title>xref-keyref-test</title>
  <topicref href="topic-a.dita" keys="topic-a"/>
  <topicref href="topic-b.dita" keys="topic-b"/>
  <topicref href="topic-c.dita" keys="topic-c"/>
</map>
```

```
<concept id="topic-a">
    <title>Topic A</title>
  <shortdesc>I am topic A.</shortdesc>
  <conbody>
    <p>I link to <xref keyref="topic-b"/> which has a link in the short
    description.</p>
  </conbody>
</concept>
```

```
<concept id="topic-b">
  <title>Topic B</title>
  <shortdesc>I am topic B with a link to <xref keyref="topic-c"/> in the short
    description.</shortdesc>
</concept>
```

```
<concept id="topic-c">
    <title>Topic C</title>
    <shortdesc>I am topic C.</shortdesc>
  <conbody>
    <p>I link to <xref keyref="topic-a"/> which has no link in the short
      description.</p>
  </conbody>
</concept>
```
# LOG
[DOTX031E]: The file  topic-c.dita  is not available to resolve link information.

# OUTPUT
All links are generated, but the link in Topic A has an incomplete `@title` attribute because the link in the short description is not resolved:

```
<a class="xref" href="topic-b.html" title="I am topic B with a link to in the short description.">Topic B</a>
``` 

# NOTES
Tested for HTML5 output. 
Identical results when I change`@keyref` addressing to direct `@href` addressing.

**Comment by raducoravu** [comment]
You should try using a newer DITA OT, ideally the latest one, there was a fix for a similar issue in DITA OT 2.3.2:
https://github.com/dita-ot/dita-ot/issues/1599
Another issue related to this:
https://github.com/dita-ot/dita-ot/issues/1786

**Comment by robander** [comment]
Also likely related to #2758
